### PR TITLE
will not exit when no kubeconfig found

### LINKED ISSUE #547: synopsysctl kube-mock feature shouldn't require a kubeconfig file
**What happened**:
ERRO[0000] stat /Users/jeremyd/.kube/config: no such file or directory

**How to reproduce it (as minimally and precisely as possible)**:
mv ~/.kube/config ~/.kube/config.backup
synopsysctl create blackduck --mock-kube yaml --admin-password test --postgres-password test --user-password test test

### PR DESCRIPTION
**Pull request by yashbhutwala**
/closes #547 

also related to issue #258 and PR #425 

@jdartigalongue @rrati

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/581#" title="Author self-approved">yashbhutwala</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [yashbhutwala]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@yashbhutwala: once the present PR merges, I will cherry-pick it on top of release-2019.6.x in a new PR and assign it to you.

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/581#issuecomment-506035431):

>/cherrypick release-2019.6.x


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: pkg/synopsysctl/cmd_root.go [hunk 1]
```diff
@@ -61,8 +61,7 @@ var rootCmd = &cobra.Command{
 		}
 		// Sets kubeconfig and initializes resource client libraries
 		if err := setResourceClients(); err != nil {
-			log.Error(err)
-			os.Exit(1)
+			log.Warnf("resource clients were not set properly, %v", err)
```
Comment (rrati) [inline pkg/synopsysctl/cmd_root.go]: I don't think we want to log anything in the case of the native command.  Printing even a warning would seem confusing imo.  May need to take most of PreRunE and place in a function that only non-native commands would use unless there is a better way to have the native sub-commands not use this PreRunE (which I'm not sure there is)

**Comment by jdartigalongue** [comment]
@yashbhutwala @rrati  I think we want to exit by default if there is no kube config file... just not when using the native command.

File: pkg/synopsysctl/cmd_root.go (hunk 1 above)
Comment (yashbhutwala) [reply pkg/synopsysctl/cmd_root.go]: There is a hack where I can have a `PersistentRunE` for a subcommand that will override the root command, but it's being fixed in cobra right now.  See issue [252](https://github.com/spf13/cobra/issues/252)

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/581#pullrequestreview-255319401" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/581#" title="Author self-approved">yashbhutwala</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,yashbhutwala]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/581#pullrequestreview-255319401" title="Approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/581#pullrequestreview-255323757" title="LGTM">rrati</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/581#" title="Author self-approved">yashbhutwala</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,rrati,yashbhutwala]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@rrati: once the present PR merges, I will cherry-pick it on top of release-2019.6.x in a new PR and assign it to you.

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/581#issuecomment-506405901):

>/cherrypick release-2019.6.x


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by operator-r2d2** [comment]
@yashbhutwala: new pull request created: #591

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/581#issuecomment-506035431):

>/cherrypick release-2019.6.x


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**